inc/zoo/swar/SWAR.h

```diff
@@ -62,13 +74,53 @@ struct SWAR {
         AllOnes = ~std::make_unsigned_t<T>{0} >> PaddingBitsCount, // Also constructed in RobinHood utils: possible bug?
         LeastSignificantBit = meta::BitmaskMaker<T, std::make_unsigned_t<T>{1}, NBits>::value,
         MostSignificantBit = LeastSignificantBit << (NBits - 1),
-        LeastSignificantLaneMask =
-            sizeof(T) * 8 == NBits ? // needed to avoid shifting all bits
-                ~T(0) :
-                ~(~T(0) << NBits),
+        LeastSignificantLaneMask = []() {
+            if constexpr (NBits < sizeof(T) * 8) {
+                return (T(1) << NBits) - 1;
+            } else {
+                return ~T(0);
+            }
+        }(),
         // Use LowerBits in favor of ~MostSignificantBit to not pollute
         // "don't care" bits when non-power-of-two bit lane sizes are supported
-        LowerBits = MostSignificantBit - LeastSignificantBit;
+        LowerBits = MostSignificantBit - LeastSignificantBit,
+        MaxUnsignedLaneValue = LeastSignificantLaneMask;
+
+    template <typename InputIt>
```

Comment:
Resolved the other comment, the aspect remaining is whether input iterators is meant here.
If you want to mean "iterator to the input", what is meant is the input iterator category
https://en.cppreference.com/w/cpp/named_req/InputIterator
This code can indeed work with input iterators.  I'm asking if you noticed that `InputIt` has a particular meaning.

Reply:
i didn't notice. did you have another idea in mind?

Reply:
Is this comment addressed? I think Eduardo's response is up.